Feature 2 Timetable Generator

JD-Grimm opened the pull request:
The test case for timetable generator does not run properly as one of the methods in the section entities isn't fully implemented yet. 

Used clean architecture.
Added timetable entity, which stores courses and has an assorted list of methods.
Added timetablegenetatorinteractor, which sets up a timetable generation.
Made test cases for both.

michaelhsj committed:
Introduced documentation and testing classes for retrieve timetable classes.

michaelhsj committed:
Implemented equals methods for the models and finished testing for the retrieve timetable use case.

Kai15116 committed:
Fixed the errors for displaying EditTimetableScreen

Kai15116 committed:
Made the RecommendBRTestMain simpler

Kai15116 committed:
Fixed error where color of the block is changed when course is added/removed from EditTimetableScreen

Kai15116 committed:
Removed System.out.println from TimetableView

JD-Grimm committed:
Merge pull request #51 from CSC207-2022F-UofT/main

Merged main into feature 2 branch

Kai15116 committed:
Added documentation to MainUI

Kai15116 committed:
Merge branch 'fix-errors-seen-in-recommendbrtestmain' of https://github.com/CSC207-2022F-UofT/course-project-todo-find-a-team-name into feature-9-finish-main-menu-ui

Kai15116 committed:
Merge branch 'feature-12-finish-timetable-editing' of https://github.com/CSC207-2022F-UofT/course-project-todo-find-a-team-name into feature-9-finish-main-menu-ui

Kai15116 committed:
Added functionality to display, edit, and generate timetable button

Kai15116 committed:
Merge branch 'feature-11-cleaning-up-retrieve-timetable' of https://github.com/CSC207-2022F-UofT/course-project-todo-find-a-team-name into feature-10-finish-timetable-view

Kai15116 committed:
Merge branch 'feature-12-finish-timetable-editing' of https://github.com/CSC207-2022F-UofT/course-project-todo-find-a-team-name into feature-10-finish-timetable-view

Kai15116 committed:
Added functionality to go back button in TimetableUI

Kai15116 committed:
Merge branch 'feature-10-finish-timetable-view' of https://github.com/CSC207-2022F-UofT/course-project-todo-find-a-team-name into feature-9-finish-main-menu-ui

Kai15116 committed:
Changed main in MainUI so previous panel is set for TimetableUI

Kai15116 committed:
Added functionality to edit button in TimetableUI

Kai15116 committed:
Added helper function for changing screen

Kai15116 committed:
Added documentation to the helper function in MainUI

moonsdust committed:
Merge pull request #53 from CSC207-2022F-UofT/main

Updated feature 6 branch

moonsdust committed:
Added FileImportInputBoundary, Updated SessionGateway & its interactor, and tests

moonsdust committed:
Deleted FileImportInputBoundary

moonsdust committed:
Updated SessionGatewayInteractor.java

moonsdust committed:
Updated SessionGateway.java with comment

JD-Grimm committed:
Changed timetable entity to check for conflicts between a timetableCourse and a timetable, and added in the timetable generator use case

moonsdust committed:
updated classes to consider FileImportRequestModel and created FileImportInputBoundary

moonsdust committed:
renamed files to camelCase

michaelhsj committed:
Updated testing documentation.

moonsdust committed:
Included docstrings in tests I wrote (related to Session Gateway, its Interactor, and Session entity)

JD-Grimm committed:
Changed timetable entity to check for conflicts between a timetableCourse and a timetable, and added in the timetable generator use case

moonsdust committed:
Created packages and removed tutorial stuff

hanshhh committed:
Implement Help Instruction Screen

moonsdust committed:
Reverted file names

michaelhsj committed:
Updated the RetrieveTimetableInteractor such that timetables and sessions can be publically set.

michaelhsj committed:
Updated the RetrieveTimetableInteractor such that timetables and sessions can be publically set.

michaelhsj committed:
Merge remote-tracking branch 'origin/feature-11-cleaning-up-retrieve-timetable' into feature-11-cleaning-up-retrieve-timetable

# Conflicts:
#	src/main/java/retrieve_timetable_use_case/RetrieveTimetableInputBoundary.java
#	src/test/java/retrieve_timetable_use_case/BlockModelTest.java

Kai15116 committed:
Added resources folder

moonsdust committed:
Updated file names and removed two methods in SessionGateway and SessionGatewayInteractor

moonsdust committed:
Updated file names and removed two methods in SessionGateway and SessionGatewayInteractor

Kai15116 committed:
Updated screen size

michaelhsj committed:
Relocated all files to follow restructured directories.

Kai15116 committed:
Organized files

moonsdust committed:
Reverted changes and added back the two methods in SessionGateway and SessionGatewayInteractor

moonsdust committed:
Updated SessionGatewayInteractorTest

moonsdust committed:
Removed unused classes and updated allSessionBuilder so it returns a HashMap instead of SessionStorer

moonsdust committed:
Reverted ArrayLists to previous location in readFromFile, added test_session_data.json, updated tests

moonsdust committed:
Reverted ArrayLists to previous location in readFromFile, added test_session_data.json, updated tests

moonsdust committed:
Moved JSON file into resources

moonsdust committed:
Combined fileToString and readFromFile

Kai15116 committed:
Fixed error in BRRecommender

moonsdust committed:
Changed readFromFile input and output to Session

Kai15116 committed:
Merge branch 'fix-errors-seen-in-recommendbrtestmain' of https://github.com/CSC207-2022F-UofT/course-project-todo-find-a-team-name into feature-5-breadth-requirement-recommendation-clean

Kai15116 committed:
Resolved most of the intelliJ warnings except for unused declaration

hanshhh committed:
Test files

hanshhh committed:
Merge branch 'file-structure-and-documentation' of https://github.com/CSC207-2022F-UofT/course-project-todo-find-a-team-name into 7-feature-7-blacklistwhitelist-domenttest

Kai15116 committed:
Implemented factory design pattern

Kai15116 committed:
Used extract method to solve long method code smell

hanshhh committed:
Test files

Kai15116 committed:
Merge branch 'feature-6-file-io' of https://github.com/CSC207-2022F-UofT/course-project-todo-find-a-team-name into feature-5-breadth-requirement-recommendation-clean

Kai15116 committed:
Updated RecommendBRTestMain so it uses session gateway

hanshhh committed:
Test files

hanshhh committed:
Test files

hanshhh committed:
Merge branch 'feature-6-file-io' of https://github.com/CSC207-2022F-UofT/course-project-todo-find-a-team-name into 7-feature-7-blacklistwhitelist-domenttest

Kai15116 committed:
Extracted method to resolve long method

Kai15116 committed:
Merge branch 'feature-5-breadth-requirement-recommendation-clean' of https://github.com/CSC207-2022F-UofT/course-project-todo-find-a-team-name into feature-10-finish-timetable-view

Kai15116 committed:
Merge branch 'feature-5-breadth-requirement-recommendation-clean' of https://github.com/CSC207-2022F-UofT/course-project-todo-find-a-team-name into feature-9-finish-main-menu-ui

Kai15116 committed:
Extracted method to resolve long method

Kai15116 committed:
Extracted method to resolve long method

Kai15116 committed:
Merge branch 'feature-9-finish-main-menu-ui' of https://github.com/CSC207-2022F-UofT/course-project-todo-find-a-team-name into feature-10-finish-timetable-view

hanshhh committed:
Created TestCases for BlacklistWhitelist UseCase Interactor

Kai15116 committed:
Extracted method to resolve long method

Kai15116 committed:
Implemented ActionListener to resolve long method

Kai15116 committed:
CHanged equals so it disregards the order of sections/blocks

Kai15116 committed:
Merge branch 'feature-11-cleaning-up-retrieve-timetable' of https://github.com/CSC207-2022F-UofT/course-project-todo-find-a-team-name into feature-5-breadth-requirement-recommendation-clean

 Conflicts:
	src/main/java/recommend_br_use_case/interface_adapters/RecommendBRPresenter.java

Kai15116 committed:
Cleaned imports

hanshhh committed:
Create Test for SectionFilterInteractor

hanshhh committed:
Fixed Dead method code smell

hanshhh committed:
Fixed Dead method code smell

hanshhh committed:
Create testCases for presenter in SectionFilterPresenterTest.java

moonsdust committed:
Refactored classes into specific packages

hanshhh committed:
Fixing code smell.

Kai15116 committed:
Merge branch 'feature-11-cleaning-up-retrieve-timetable' of https://github.com/CSC207-2022F-UofT/course-project-todo-find-a-team-name into feature-10-finish-timetable-view

 Conflicts:
	src/main/java/recommend_br_use_case/interface_adapters/RecommendBRPresenter.java

Kai15116 committed:
Implemented Controller/intractor/presenter for displaying timetable UI

Kai15116 committed:
Organized code

michaelhsj committed:
Implemented Flow.Subscriber in the retrieve timetable interactor.

michaelhsj committed:
Merge remote-tracking branch 'origin/feature-11-cleaning-up-retrieve-timetable' into feature-11-cleaning-up-retrieve-timetable

Kai15116 committed:
Added comments

Kai15116 committed:
Removed argument for the constructor in DisplayTimetablePresenter

JD-Grimm committed:
Merge pull request #63 from CSC207-2022F-UofT/main

Merge main to feature 2

JD-Grimm committed:
Merge remote-tracking branch 'origin/8-feature-2-generating-with-course-list' into 8-feature-2-generating-with-course-list

Kai15116 committed:
Removed argument for the constructor in RecommendBRPresenter

Kai15116 committed:
Removed redundant methods

JD-Grimm committed:
Added timetabletest and timetablegeneratortest

hanshhh committed:
Edit the main

moonsdust reviewed (changes requested):
Your code looks good! I also can see that you implemented clean architecture as seen with you using an interface, request model, and interactor. Although it seems that your interactor is returning an entity, which it shouldn't do since it would violate clean architecture. I would like to add that you should keep an eye on when the method that you used from the section entity to see when it has been fixed so you can confirm the correctness of your methods and then I can approve your pull request!

Kai15116 reviewed (changes requested):
I found some issues that you might want to address: 
- Controller/Presenter seems to be missing
- Interactor should only accept request model and should only return response model, it should not accept/return entity directly since clean architecture will be violated when calling from the controller
- RequestModel cannot contain a reference to entities because this would also violate clean architecture
- Method name should usually be a verb (e.g. timetableGenerator should be generateTimetable to avoid confusion)

Ideally, I think you should add more documentation about the summary of your feature and any design decision you made.

moonsdust committed:
Updated classes to return SessionModel (for interactor) and changed SessionBuilderInteractor to SessionBuilder. Also updated tests

michaelhsj committed:
Replaced RT interactor's constructor with a no-argument constructor.

michaelhsj committed:
Replaced RT interactor's constructor with a no-argument constructor.

JD-Grimm committed:
Added timetabletest and timetablegeneratortest
Modified my use case to better fit with other use cases. Fixed code structure to follow clean architecture better.

Kai15116 committed:
Merge branch 'feature-5-breadth-requirement-recommendation-clean' of https://github.com/CSC207-2022F-UofT/course-project-todo-find-a-team-name into feature-9-finish-main-menu-ui

Kai15116 committed:
Modified to match the pulled branch

YahyaElgabra committed:
set up most if not all files, still requires including flow, actually sorting, and integration

YahyaElgabra committed:
Merge pull request #72 from CSC207-2022F-UofT/feature-11-cleaning-up-retrieve-timetable

Feature 11 to Feature 13

YahyaElgabra committed:
cleaning, worked on UI

YahyaElgabra committed:
working on updating timetables in AllTimetablesScreen

YahyaElgabra committed:
Merge remote-tracking branch 'origin/feature-13-finishing-sorter' into feature-13-finishing-sorter

# Conflicts:
#	src/main/java/display_timetable_use_case/interface_adapters/TimetableView.java
#	src/main/java/screens/SessionViewModel.java

YahyaElgabra committed:
Merge pull request #73 from CSC207-2022F-UofT/feature-10-finish-timetable-view

Feature 10 to feature 13

YahyaElgabra committed:
integrated AllTimetablesScreen with TimetablesSortMenu (theoretically, it should work but doesnt)

YahyaElgabra committed:
Merge remote-tracking branch 'origin/feature-9-finish-main-menu-ui' into feature-13-finishing-sorter

JD-Grimm committed:
Added timetabletest and timetablegeneratortest
Modified my use case to better fit with other use cases. Fixed code structure to follow clean architecture better.

JD-Grimm committed:
Merge remote-tracking branch 'origin/feature-6-file-io' into 8-feature-2-generating-with-course-list

hanshhh committed:
Edit the UI in a way that connects with UseCase 2

hanshhh committed:
Fixed the Clean Architecture

YahyaElgabra committed:
fix frame.pack bug.... took hours of my time

michaelhsj committed:
Simplified RTInputBoundary methods.

YahyaElgabra committed:
Merge remote-tracking branch 'origin/7-feature-7-blacklistwhitelist-domenttest' into feature-13-finishing-sorter

hanshhh committed:
Remove extra Session Entity.

hanshhh committed:
fixed minor typo

Kai15116 committed:
Added generateTimetableResponse in EntityConverter

YahyaElgabra committed:
in the middle of implementing interactor, this commit is to have most updated version for others to pull

Kai15116 committed:
Merge branch '7-feature-7-blacklistwhitelist-domenttest' of https://github.com/CSC207-2022F-UofT/course-project-todo-find-a-team-name into feature-13-finishing-sorter

Kai15116 committed:
Resolved errors in MainUI

JD-Grimm committed:
Used Kai's workaround for a cleaner generator and to help interface with others. Using Hans's screen for a direct dependency with his use case

JD-Grimm committed:
Merge remote-tracking branch 'origin/feature-11-cleaning-up-retrieve-timetable' into 8-feature-2-generating-with-course-list

# Conflicts:
#	src/main/java/edit_timetable_use_case/AddCourseInteractor.java

JD-Grimm committed:
Merge remote-tracking branch 'origin/feature-13-finishing-sorter' into 8-feature-2-generating-with-course-list

# Conflicts:
#	src/main/java/RecommendBRTestMain.java

JD-Grimm committed:
fixed import statements

JD-Grimm committed:
Added two more classes

moonsdust reviewed (commented):
This looks good now and good use of clean architecture (previously you were returning entities from the interactor, which had violated CA)! Would suggest adding some comments to InputBoundary and your public methods but overall it looks a lot better now
